src/lib/y2users/linux/writer.rb

```diff
@@ -100,6 +101,8 @@ def write
         issues = Y2Issues::List.new
 
         add_groups(issues)
+        # Configure useradd after creating the groups, to make sure the default group exists already
```

Comment:
So add_groups reads the gid of newly created group? Otherwise it won't work.

Reply:
Why? Useradd will use that group by default in those cases in which the Ruby object does not provide a group. So this is only relevant for useradd itself, who reads the users from the system and has no visibility on the list of groups we have in our Ruby objects. It's somehow irrelevant for our code.

Reply:
OK, lets consider this scenario:

- default group specified by group name in autoyast profile
- group does not have gid in autoyast profile
- group does not exist yet on system

So what we need to do in such case is to create group, read gid it gets assigned and use it in this useradd_config...or does it support also group name instead of gid as argument?

Reply:
Yes. Useradd supports both group name and gid for the GROUP configuration key. Actually, what AutoYaST does is copying the value from the profile as-is (be it a number or a name) to the useradd configuration. No need to do any conversion.

Reply:
ok, then it is fine for me. Just this comment confuse me as it looks like it is mandatory to have group existing before configuring useradd, which I am not sure if it is case. I think it is just needed before calling useradd when creating user, not?

Reply:
Yes, I can rephrase it a bit

Reply:
Although it's actually true that the group must exists before configuring useradd. Otherwise useradd will complain like this:

`
$ useradd -D --gid not_a_group
useradd: group 'not_a_group' does not exist
`